src/core/utils/misc.cc

```diff
@@ -23,16 +23,16 @@
 #include "utils/misc.h"
 #include <stdint.h>
 #include <cstring>    // std::memcpy
-
-
 namespace dt {
 
+
 // The warning is spurious, because shifts triggering the warning are
 // within if() conditions that get statically ignored.
-#if DT_COMPILER_GCC
-  #pragma GCC diagnostic push
-  #pragma GCC diagnostic ignored "-Wshift-count-overflow"
-#endif
+DISABLE_GCC_WARNING("-Wshift-count-overflow")
+// 4293: shift count negative or too big, undefined behavior
+DISABLE_MSVC_WARNING(4293)
+// 4333: right shift by too large amount, data loss
+DISABLE_MSVC_WARNING(4333)
```

Comment:
Where do we restore these?

Reply:
Technically we don't need to worry about restoring warnings at the end of translation unit, since the compiler exits at that point anyways.
But I can restore them if you want.

Reply:
Yeah, but originally these warnings were silenced only for several lines of code, and now they're silenced for the whole translation unit. Who knows how the code in `misc.cc` will evolve, so why not to silence the warnings only on the required lines?